### PR TITLE
style: improve Drawer appear/leave motion

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- 优化了 Drawer 进出场动画效果，更加简洁不拖沓。原版动画的内容移动距离过长，感觉上会有点闪屏。效果参考：https://pocketbase.io/demo/ （Try to click New Record button）

- Old：https://ant.design/components/drawer-cn

Jan-09-2024 17-05-24

- New：https://preview-46881-ant-design.surge.sh/components/drawer-cn

https://github.com/ant-design/ant-design/assets/507615/4f0531bc-767d-4057-9183-6481e7bec539

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Improve Drawer appear/leave motion style.        |
| 🇨🇳 Chinese | 优化 Drawer 进出场动画效果，使其更加简洁不拖沓。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
Safari 下 Drawer 动画有问题，还得修一下。

**Comment by afc163** [comment]
压缩真是一个奇妙的事情。。。

**Review by crazyair — DISMISSED** [review]

**Comment by afc163** [comment]
图片
还有问题。

File: components/drawer/style/motion.ts [hunk 1]
```diff
@@ -1,132 +1,76 @@
...
-            '&-leave': {
-              transform: 'translateY(0)',
-              '&-active': {
-                transform: 'translateY(100%)',
-              },
-            },
-          },
-        ],
-      },
+      [`${componentCls}-panel-motion`]: ['left', 'right', 'top', 'bottom'].reduce(
```
Comment (crazyair) [inline components/drawer/style/motion.ts:67]: as const 或 as Direction
```tsx
[`${componentCls}-panel-motion`]: (['left', 'right', 'top', 'bottom'] as const).reduce(
        (obj, direction) => ({
          ...obj,
          [`&-${direction}`]: getPanelMotionStyles(direction, motionDurationSlow),
        }),
        {},
      ),
```
这样有类型校验

image

Comment (afc163) [inline components/drawer/style/motion.ts:67]:
```suggestion
      [`${componentCls}-panel-motion`]: (['left', 'right', 'top', 'bottom'] as const).reduce(
```

File: components/drawer/style/motion.ts [hunk 2]
```diff
@@ -1,132 +1,76 @@
...
-              transform: 'translateY(0)',
-              '&-active': {
-                transform: 'translateY(100%)',
-              },
-            },
-          },
-        ],
-      },
+      [`${componentCls}-panel-motion`]: (['left', 'right', 'top', 'bottom'] as const).reduce(
+        (obj, direction: Direction) => ({
```
Comment (crazyair) [inline components/drawer/style/motion.ts:68]:
```suggestion
        (obj, direction) => ({
```

File: components/drawer/style/motion.ts [hunk 3]
```diff
@@ -61,11 +61,11 @@ const genMotionStyle: GenerateStyle<DrawerToken> = (token) => {
   return {
     [componentCls]: {
       // ======================== Mask ========================
-      [`${componentCls}-mask-motion`]: getFadeStyle(0, motionDurationSlow),
+      [`${componentCls}-mask-motion`]: getFadeStyle(motionDurationSlow),
```
Comment (afc163) [inline components/drawer/style/motion.ts:64]:
```suggestion
      [`${componentCls}-mask-motion`]: getFadeStyle(0, motionDurationSlow),
```

**Pull request closed without merging**